Make Py3 compatible. [3]

amrsoll opened the pull request:
I mainly disabled analytics by default to debug easily.

amrsoll committed:
Switch to python3.

amrsoll committed:
Deactivate analytics by default.

amrsoll committed:
Merge branch 'develop' into py3

amrsoll committed:
Bump version.

amrsoll commented:
Also I'm currently using this for BeamOS

I know the newer version of the LED driver doesn't have the spread spectrum feature, we might want to talk about this.

amrsoll committed:
Fix py2 absolute import.

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/mrbeam/MrBeamLedStrips/be6712e66356778af76f75611ceb6f9000e965d3/mrbeam/MrBeamLedStrips/edf4507a8d3b231f38138f2784cd9140606ad354/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) your code in 25.952 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/mrbeam/MrBeamLedStrips/be6712e66356778af76f75611ceb6f9000e965d3/mrbeam/MrBeamLedStrips/edf4507a8d3b231f38138f2784cd9140606ad354/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=mrbeam)

amrsoll committed:
Fix default config.

amrsoll committed:
Fix indentation.

amrsoll committed:
Merge branch 'f_default_config' into py3

amrsoll committed:
Fix import.

amrsoll committed:
Fix config import name.

andywerner reviewed (changes requested):
Please do not disable analytics! How would it get enabled?
If the user did not agree to analytics no analytics data get written nor uploaded by our plugin. But in this case it's required that the analytics data are sent to the plugin in any case.

amrsoll committed:
Merge branch 'f_default_config' into py3

amrsoll committed:
Single packet communication.

Reduces trickery to fix bad communication handling with sendall and
recv.
Also, code becomes py2 incompatible (with using str to decode utf8)

amrsoll committed:
Rm unnecessary packet sent

amrsoll committed:
2to3.

amrsoll committed:
absolute import.

andywerner reviewed (changes requested):
make it work without config file and enable analytics

amrsoll committed:
Enable analytics by default.

amrsoll commented:
TODO : create if branches / try catch to differentiate between py2 and py3, because the neopixel lib and functions have changed.
See 5d4acbd

amrsoll committed:
Reverse compat for py2.

amrsoll committed:
Fix setup.py to have pip 9

Only pip 9.x compatible for python2.

andywerner commented on mrbeam_ledstrips/client.py:
```diff
@@ -5,34 +5,33 @@
-		data = s.recv(5000)
-		print "< " + data
-		s.close()		
+		# s.sendall(b'info')
```
please check

andywerner reviewed (approved):
please look at the comment

amrsoll committed:
Remove comment.
